pkg/maps/ipmasq/ipmasq.go

```diff
@@ -86,32 +86,42 @@ type IPMasqBPFMap struct{}
 
 func (*IPMasqBPFMap) Update(cidr net.IPNet) error {
 	if ip.IsIPv4(cidr.IP) {
-		return IPMasq4Map().Update(keyIPv4(cidr), &Value{})
+		if option.Config.EnableIPv4Masquerade {
+			return IPMasq4Map().Update(keyIPv4(cidr), &Value{})
+		}
 	} else {
-		return IPMasq6Map().Update(keyIPv6(cidr), &Value{})
+		if option.Config.EnableIPv6Masquerade {
+			return IPMasq6Map().Update(keyIPv6(cidr), &Value{})
+		}
 	}
+	return nil
 }
 
 func (*IPMasqBPFMap) Delete(cidr net.IPNet) error {
 	if ip.IsIPv4(cidr.IP) {
-		return IPMasq4Map().Delete(keyIPv4(cidr))
+		if option.Config.EnableIPv4Masquerade {
+			return IPMasq4Map().Delete(keyIPv4(cidr))
+		}
 	} else {
-		return IPMasq6Map().Delete(keyIPv6(cidr))
+		if option.Config.EnableIPv6Masquerade {
+			return IPMasq6Map().Delete(keyIPv6(cidr))
+		}
 	}
+	return nil
 }
 
-func (*IPMasqBPFMap) Dump() ([]net.IPNet, error) {
+func (*IPMasqBPFMap) DumpForProtocols(ipv4Needed, ipv6Needed bool) ([]net.IPNet, error) {
 	cidrs := []net.IPNet{}
```

Comment:
nit: Prefer var cidrs []net.IPNet{} or pre-allocate if known size to ease up on GC

Reply:
Thanks Quentin! I see what your are saying. You would know best here is there is something we could use based on the expected/average number of cidrs coming back.  Then this would avoid some GC churn. But this is an optimization and I understand you're trying to push thru bugs so feel free to disregard if not a good time to do so.

Reply:
OK thank you! The only value we could use is the max size for the map, otherwise we'll get partial dumps if users ever fill their maps. Now I've got very little idea of whether the trade-off between GC churn and memory consumption would be interesting if we did so. As you say, I'll consider it beyond the scope of the PR and will leave it for another day. But thanks!